lib/beaker-hostgenerator/parser.rb

```diff
@@ -168,28 +166,26 @@ def parse_node_info_token(token)
       node_info = NODE_REGEX.match(token)
 
       if node_info
-        node_info = Hash[node_info.names.zip(node_info.captures)]
+        node_info = node_info.names.zip(node_info.captures).to_h
       else
         raise BeakerHostGenerator::Exceptions::InvalidNodeSpecError.new,
               "Invalid node_info token: #{token}"
       end
 
-      if node_info['arbitrary_roles']
-        node_info['arbitrary_roles'] =
-          node_info['arbitrary_roles'].split(',') || ''
-      else
-        # Default to empty list to avoid having to check for nil elsewhere
-        node_info['arbitrary_roles'] = []
-      end
+      node_info['arbitrary_roles'] = if node_info['arbitrary_roles']
+                                       node_info['arbitrary_roles'].split(',') || ''
+                                     else
+                                       # Default to empty list to avoid having to check for nil elsewhere
+                                       []
+                                     end
```

Comment:
I don't understand the `|| []` here. `.split(',')` alwas returns a list.

I'd make this
```suggestion
      node_info['arbitrary_roles'] = node_info.fetch('arbitrary_roles', '').split(',')
```